rs/p2p/consensus_manager/src/receiver.rs

```diff
@@ -494,21 +491,27 @@ where
                     Ok((artifact, peer_id)) => {
                         let id = artifact.id();
                         // Send artifact to pool
-                        sender.send(UnvalidatedArtifactMutation::Insert((artifact, peer_id)));
+                        if sender.send(UnvalidatedArtifactMutation::Insert((artifact, peer_id))).is_err() {
+                            error!(log, "The receiving side of the channel was closed. This should be infallible situation since a cancellation token should be received. If this happens then most likely there is very subnet synchonization bug.");
+                        }
 
                         // wait for deletion from peers
-                        peer_rx.wait_for(|p| p.is_empty()).await;
+                        // TODO: handle the error appropriately
+                        let _ = peer_rx.wait_for(|p| p.is_empty()).await;
 
                         // Purge from the unvalidated pool
-                        sender.send(UnvalidatedArtifactMutation::Remove(id));
+                        if sender.send(UnvalidatedArtifactMutation::Remove(id)).is_err() {
+                            error!(log, "The receiving side of the channel was closed. This should be infallible situation since a cancellation token should be received. If this happens then most likely there is very subnet synchonization bug.");
+                        }
```

Comment:
Could be worth mentioning what channel it is that was closed on the receiving end to have more information when reading the log in Kibana.

Reply:
Also, could you assign the error message to a variable, instead of duplicating it?